### PR TITLE
[container]: Refactor setup container script

### PR DESCRIPTION
**Pull request by nazariig**
### Description of PR

**Summary:**
This is a refactoring/bugfixing/improvement of the  `setup-container.sh`

### Type of change

- [X] Bug fix
- [X] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
* Bugfixes & improvements for `setup-container.sh` script

#### How did you do it?
* Reworked existing stuff & added new features

#### How did you verify/test it?
1. ./setup-container.sh -n sonic-mgmt-nazariig_master

#### Any platform specific information?
* N/A

#### Supported testbed topology if it's a new test case?
* N/A

### Documentation 

```bash
nazariig@sonic: sonic-mgmt$ ./setup-container.sh -h
Usage ./setup-container.sh [OPTIONS]

Mandatory options:
  -n <container_name>  set the name of the Docker container

Other options:
  -i <image_id>        specify Docker image to use. This can be an image ID (hashed value) or an image name.
                       If no value is provided, defaults to the following images in the specified order:
                         1. The local image named "docker-sonic-mgmt"
                         2. The local image named "sonicdev-microsoft.azurecr.io:443/docker-sonic-mgmt"
                         3. The remote image at "sonicdev-microsoft.azurecr.io:443/docker-sonic-mgmt"
  -d <directory>       specify directory inside container to bind mount to sonic-mgmt root (default: "/var/src")
  -m <mount_point>     specify directory to bind mount to container
  -p <port>            publish container port to the host
  -f                   automatically remove the container when it exits
  -v                   explain what is being done
  -h                   display this help and exit

Examples:
  ./setup-container.sh -n sonic-mgmt-nazariig_master
  ./setup-container.sh -n sonic-mgmt-nazariig_master -i sonicdev-microsoft.azurecr.io:443/docker-sonic-mgmt:latest
  ./setup-container.sh -n sonic-mgmt-nazariig_master -d /var/src
  ./setup-container.sh -n sonic-mgmt-nazariig_master -m /my/working/dir
  ./setup-container.sh -n sonic-mgmt-nazariig_master -p 192.0.2.1:8080:80/tcp
  ./setup-container.sh -h
```

### CODE REVIEW

**Comment by nazariig** [comment]
@lguohan / @theasianpianist / @liushilongbuaa please have a look

File: setup-container.sh [hunk 1]
```diff
@@ -1,176 +1,377 @@
...
+# Permissions configuration
+RUN usermod -a -G sudo {{ USER_NAME }}
+RUN usermod -a -G {{ DGROUP_NAME }} {{ USER_NAME }}
+RUN echo '{{ USER_NAME }} ALL=(ALL) NOPASSWD:ALL' > /etc/sudoers.d/{{ USER_NAME }}
+RUN chmod 0440 /etc/sudoers.d/{{ USER_NAME }}
+RUN chown -R '{{ USER_ID }}:{{ GROUP_ID }}' /home/{{ USER_NAME }}
+
+# SSH/PASS configuration
+RUN sed -i -E 's/^#?PermitRootLogin.*$/PermitRootLogin yes/g' /etc/ssh/sshd_config
+RUN echo 'root:{{ ROOT_PASS }}' | chpasswd
```
Comment (theasianpianist) [inline setup-container.sh:222]: What is the use case for adding root login?

**Review by theasianpianist — APPROVED** [review]
Thanks for the improvements! Just left a quick question about adding root login

File: setup-container.sh (hunk 1 above)
Comment (nazariig) [reply setup-container.sh:222]: @theasianpianist this is done for management purposes only: on some older docker versions there is no automatic terminal adjusting according to the window size, so using the exec is a bit painful. This is a reason of having SSH for root.

**Pull request merged**